### PR TITLE
For lin grad, clarify effect of rot'n vector if angle is >90

### PR DESCRIPTION
**Pull request by PeterConstable**
This is proposed text to address #196

### CODE REVIEW

**Comment by anthrotype** [comment]
maybe instead of (or if you like in addition to) rotation and angles, you could mention this alternative way which I find useful to think about a transformed linear gradient, which I hinted at in https://github.com/googlefonts/colr-gradients-spec/issues/196#issuecomment-758817422

Basically, when given the three P0, P1 and P2 you want to find the point (shal we call it P3) that gives you the _effective_ linear gradient between P0 and P3 which is identical and functionally equivalent to the one between P0 and P1 as rotated by P2.

The point P3 is the orthogonal projection of P1 onto the line between P0 and P2.
P0, P1, and P2 describe two vectors: `v = P1 - P0` and `u = P2 - P0`. You compute the [vector projection](https://en.wikipedia.org/wiki/Vector_projection) of `v` onto `u` -- which returns another vector -- and add that to P0 to obtain P3.

**Comment by PeterConstable** [comment]
That wording gets complicated and is a bit hard to follow. But there's a related idea that might be useful: for each position along the colour line (positioned on p0p1), the colour at that position is projected to the sides in a direction orthogonal to the line p0p2.

**Comment by anthrotype** [comment]
yes, I like that. It's also consistent with my algorithm for finding the implied P3 projection

**Comment by anthrotype** [comment]
however, in the end an implementor reading this spec really wants to find a way to reduce these triplet P0, P1 and P2 to two points which describe the start and end position of the final linear gradient, that they can plug in a graphics API that  mostly already knows how to draw a linear gradient between two points..

**Comment by PeterConstable** [comment]
> 
> 
> however, in the end an implementor reading this spec really wants to find a way to reduce these triplet P0, P1 and P2 to two points which describe the start and end position of the final linear gradient, that they can plug in a graphics API that mostly already knows how to draw a linear gradient between two points..

On the one hand, the ability to express an equivalent gradient using only one vector begs the question as to why we specify it using two. On the other hand (and this is a partial response to that question), the vector p0p3 wouldn't give the placement of the colour line: stop offset 0 aligns with p0, but stop offset 1 doesn't align with p3.

**Comment by rsheeter** [comment]
The proposed wording feels like it calls [p0 p2] a vector then strips away it's vector'ness. Describing it in terms of how to compute a point P3 such that you can draw a linear gradient P0=>P3, compute colors along it and project those colors orthogonal to the line, seems like it might be clearer and more helpful to implementers?

Such an explanation will likely demand explanation of why our linear gradient isn't directly defined that way. I believe the answer was to enable variation scenarios where the points tracked other points, such as points in geometry?

> stop offset 1 doesn't align with p3.

you want it to iiuc

**Comment by PeterConstable** [comment]
>Such an explanation will likely demand explanation of why our linear gradient isn't directly defined that way. I believe the answer was to enable variation scenarios where the points tracked other points, such as points in geometry?

Yup, I raised that question before. Others will likely wonder.

I've rev'd the wording, adding a note describing the p0p3 vector as Cosimo suggested, and also anticipating the above question.

File: OFF_AMD2_WD.md [hunk 1]
```diff
@@ -284,16 +284,29 @@ stops, and set the extend mode to *reflect*.
...
 
 An additional point, p₂, is also used to rotate the gradient orientation in the
-space on either side of the line defined by p₀ and p₁. The vector from p₀ to p₂
-can be referred to as the *rotation vector*. If the rotation vector is colinear
-with the line p₀p₁, there is no rotation: colors in the space on either side of
-the line p₀p₁ extend in the perpendicular direction. But if the rotation vector
-is not colinear, the gradient is drawn skewed by the angle between p₀p₁ and
-p₀p₂.
+space on either side of the line defined by p₀ and p₁. The line passing through points p₀ and p₂
+determine the direction in which the gradient progresses. If the the line p₀p₂ is colinear with the line
```
Comment (vlevantovsky) [inline OFF_AMD2_WD.md:294]: A minor nit for clarity (only because I am not in favor of using the term "colinear" in reference to lines): 

If the points p₀, p₁ and p₂ are colinear, there is no rotation: color in the space on either side of the line p₀p₁
extend in the perpendicular direction. But if the points p₀, p₁ and p₂ are not colinear, the gradient is drawn skewed by the acute angle (less than 90°) between the lines defined by p₀p₁ and by p₀p₂.

Comment (PeterConstable) [reply OFF_AMD2_WD.md]: I've revised further to address this concern.

Comment (vlevantovsky) [reply OFF_AMD2_WD.md]: Thank you.

**Comment by rsheeter** [comment]
This looks OK to me but I'd like @anthrotype or @drott to double-check as they've spent more time knee deep in the gradients.

**Review by rsheeter — APPROVED** [review]

File: OFF_AMD2_WD.md [hunk 2]
```diff
@@ -284,16 +284,29 @@ stops, and set the extend mode to *reflect*.
...
 
 An additional point, p₂, is also used to rotate the gradient orientation in the
-space on either side of the line defined by p₀ and p₁. The vector from p₀ to p₂
-can be referred to as the *rotation vector*. If the rotation vector is colinear
-with the line p₀p₁, there is no rotation: colors in the space on either side of
-the line p₀p₁ extend in the perpendicular direction. But if the rotation vector
-is not colinear, the gradient is drawn skewed by the angle between p₀p₁ and
-p₀p₂.
+space on either side of the line p₀p₁. The line passing through points p₀ and p₂
+(line p₀p₂) determine the direction in which the gradient progresses. If p₂ is
```
Comment (anthrotype) [inline OFF_AMD2_WD.md:294]: maybe "orientation" would be better here than "direction"? the latter makes me think of an arrow or vector.

**Review by anthrotype — APPROVED** [review]
LGTM thanks!

File: OFF_AMD2_WD.md (hunk 2 above)
Comment (anthrotype) [reply OFF_AMD2_WD.md:294]: never mind, it's ok like this

**Review by drott — APPROVED** [review]
LGTM, too.
